third-party/cmake/DownloadAndExtractFile.cmake

```diff
@@ -59,42 +59,47 @@ string(REPLACE ";" "-" fname "${fname}")
 set(file ${DOWNLOAD_DIR}/${fname})
 message(STATUS "file: ${file}")
 
-message(STATUS "downloading...
-     src='${URL}'
-     dst='${file}'
-     timeout='${timeout_msg}'")
-
-file(DOWNLOAD ${URL} ${file}
-  ${timeout_args}
-  ${hash_args}
-  STATUS status
-  LOG log)
-
-list(GET status 0 status_code)
-list(GET status 1 status_string)
-
-if(NOT status_code EQUAL 0)
-  # Retry on certain errors, e.g. CURLE_COULDNT_RESOLVE_HOST, which is often
-  # seen with libtermkey (www.leonerd.org.uk).
-  if((status_code EQUAL 6)  # "Couldn't resolve host name"
-    OR (status_code EQUAL 7))  # "Couldn't connect to server"
-    message(STATUS "warning: retrying '${URL}' (${status_string}, status ${status_code})")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E sleep 10)
-    file(DOWNLOAD ${URL} ${file}
-      ${timeout_args}
-      ${hash_args}
-      STATUS status
-      LOG log)
-    list(GET status 0 status_code)
-    list(GET status 1 status_string)
-  endif()
-  if(NOT status_code EQUAL 0)
-    message(FATAL_ERROR "error: downloading '${URL}' failed
-  status_code: ${status_code}
-  status_string: ${status_string}
-  log: ${log}
-")
-  endif()
+if(EXISTS ${file})
```

Comment:
This will cause the download to never happen if the file wasn't downloaded yet?

Reply:
silly me, I've fixed it